### ISSUE TITLE
+lang/scala MELPA scala-mode2 not found

### ISSUE THREAD

**Issue by bowerth** [syl20bnr/spacemacs#6100]
#### Description
The scala-mode2 package listed in the +lang/scala layer cannot be found on melpa

#### Reproduction guide
- Start Emacs
- edit .spacemacs file SPC f e d
- add `scala` to `dotspacemacs-configuration-layers`

*Observed behaviour:*
Found 1 new package(s) to install...
--> refreshing package archive: gnu... [3/3]
--> installing package: scala-mode2@scala... [1/1]
Package scala-mode2 is unavailable. Is the package name misspelled?

*Expected behaviour:*
Installation of scala-mode2 package succeeds

#### System Info
- OS: windows-nt
- Emacs: 25.0.93.1
- Spacemacs: 0.105.20
- Spacemacs branch: master (rev. 8c0a86f)
- Graphic display: t
- Distribution: spacemacs
- Editing style: emacs
- Completion: helm
- Layers:
```elisp
(c-c   clojure emacs-lisp ess eyebrowse git go latex markdown mu4e org polymode python scala yaml)
```

**Comment by syohex** [comment]
https://github.com/melpa/melpa/pull/3906

scala-mode2 is dead.

**Comment by syl20bnr** [comment]
Fixed in develop, I will release a new 0.105 this week.

**Comment by syl20bnr** [comment]
Backported to 0.105.21
